### PR TITLE
feat/even faster uploads PE 1056

### PR DESCRIPTION
**Pull request by jdaev**
PE-1056

### CODE REVIEW

File: lib/src/models/transaction_uploader.dart [hunk 1]
```diff
@@ -93,15 +93,11 @@ class TransactionUploader {
 
     if (!_txPosted) {
       await _postTransaction();
-      return;
```
Comment (arielmelendez) [inline lib/src/models/transaction_uploader.dart:96]: What's the rationale for proceeding here? Is it that _postTransaction will now retry until there's been too many failures, and so you can be assured that tx posting is done by the time the function exits?

File: lib/src/models/transaction_uploader.dart [hunk 2]
```diff
@@ -93,15 +93,11 @@ class TransactionUploader {
...
-    final newChunksCount = MAX_CHUNKS_BATCH_SIZE - _failedChunks.length;
-    final chunks = List.from(
-        _failedChunks + _transaction.getChunks(_chunkOffset, newChunksCount));
-    _failedChunks.clear();
-    _chunkOffset += newChunksCount;
+    final chunks = _transaction.getChunks(_chunkOffset, MAX_CHUNKS_BATCH_SIZE);
+    _chunkOffset += MAX_CHUNKS_BATCH_SIZE;
     try {
-      await Future.wait(chunks.map((chunk) async {
+      Future<void> uploadChunk(TransactionChunk chunk) async {
```
Comment (arielmelendez) [inline lib/src/models/transaction_uploader.dart:100]: Is it strictly necessary for this to be a nested function?

Comment (arielmelendez) [reply lib/src/models/transaction_uploader.dart:100]: It's strange to see a function declaration made within a try

File: lib/src/transactions.dart [hunk 3]
```diff
@@ -80,10 +80,11 @@ class ArweaveTransactionsApi {
   }) async* {
     final uploader = await getUploader(transaction, forDataOnly: dataOnly);
 
+    if (!dryRun) {
+      uploader.batchUploadChunks();
+    }
     while (!uploader.isComplete) {
```
Comment (arielmelendez) [inline lib/src/transactions.dart:86]: This is still sort of an infinite loop. The way is should be written is:
```
if (!dryRun) {
  while (!uploader.isComplete) {
    uploader.batchUploadChunks();
    yield uploader;
  }
}
```

Comment (jdaev) [reply lib/src/transactions.dart:86]: Updated

**Pull request closed without merging**